### ISSUE TITLE
Parameter Filter Is Not Called If Operation Was Created From OpenApi Metadata

### ISSUE THREAD

**Issue by warappa** [domaindrivendev/Swashbuckle.AspNetCore#2613]
Parameter filters are only executed if Swagger creates OpenApiOperation from scratch and not by metadata.

The `SwaggerGenerator.GenerateOperation` calls `GenerateOpenApiOperationFromMetadata(...)` to get an operation from metadata. If it got one, **no filters** **_except_ the operation filters** will be called, because those are in the sub-call-trees of `GenerateParameters(...)`, `GenerateRequestBody(...)`,...

https://github.com/domaindrivendev/Swashbuckle.AspNetCore/blob/8f363f7359cb1cb8fa5de5195ec6d97aefaa16b3/src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs#L208-L216

Please support excuting all filters regardless if there is already openapi metadata available.

**Comment by warappa** [comment]
### Workaround
Here a workaround for those who face the same problem where an **operation filter** - which gets always called - does the work instead of Swashbuckle:

```csharp
public class WorkaroundParameterFilterNotCalledOperationFilter : IOperationFilter
{
    private readonly SwaggerGenOptions _options;

    public WorkaroundParameterFilterNotCalledOperationFilter(SwaggerGenOptions options)
    {
        _options = options;
    }

    public void Apply(OpenApiOperation operation, OperationFilterContext context)
    {
        WorkaroundParameterFiltersNotBeingCalled(operation, context);
    }

    private void WorkaroundParameterFiltersNotBeingCalled(OpenApiOperation operation, OperationFilterContext context)
    {
        foreach (var descriptor in _options.ParameterFilterDescriptors)
        {
            var filter = (IParameterFilter)Activator.CreateInstance(descriptor.Type, descriptor.Arguments)!;

            foreach (var parameter in operation.Parameters)
            {
                var desc = context.ApiDescription.ParameterDescriptions.First(x => x.Name == parameter.Name);
                var t = desc.ParameterDescriptor.ParameterType;

                var parameterContext = new ParameterFilterContext(desc, context.SchemaGenerator, context.SchemaRepository, null, desc.ParameterInfo());

                filter.Apply(parameter, parameterContext);
            }
        }
    }
}
```

Register as usual:
```csharp
builder.Services.AddSwaggerGen(config =>
{
    config.EnableAnnotations();
    config.OperationFilter<WorkaroundParameterFilterNotCalledOperationFilter>(options);
    // register your parameter filter as usual
    config.ParameterFilter<YourCustomParameterFilter>();
});
```

**Comment by warappa** [comment]
Actually the same issue as in [Support ParameterFilter and RequestBodyFilter on OpenApiOperation from metadata](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues/2561).

**Comment by warappa** [comment]
Still an issue with `Swashbuckle.AspNetCore` `6.5.0` and `ASP.NET Core` `8.0.1`.

**Comment by jgarciadelanoceda** [comment]
It's added with #3059

**Comment by jgarciadelanoceda** [comment]
It's fixed with #3059
